cms/djangoapps/contentstore/rest_api/v2/views/downstreams.py

```diff
@@ -113,57 +118,78 @@ class _AuthenticatedRequest(Request):
     user: User
 
 
-# TODO: Potential future view.
-# @view_auth_classes(is_authenticated=True)
-# class DownstreamListView(DeveloperErrorViewMixin, APIView):
-#     """
-#     List all blocks which are linked to upstream content, with optional filtering.
-#     """
-#     def get(self, request: _AuthenticatedRequest) -> Response:
-#         """
-#         Handle the request.
-#         """
-#         course_key_string = request.GET['course_id']
-#         syncable = request.GET['ready_to_sync']
-#         ...
+class DownstreamListPaginator(DefaultPagination):
+    """Custom paginator for downstream entity links"""
+    page_size = 100
+    max_page_size = 1000
+
+    def paginate_queryset(self, queryset, request, view=None):
+        if 'no_page' in request.query_params:
+            return queryset
+
+        return super().paginate_queryset(queryset, request, view)
+
+    def get_paginated_response(self, data, *args, **kwargs):
+        if 'no_page' in args[0].query_params:
+            return Response(data)
+        return Response({
+            'next': self.page.next_page_number() if self.page.has_next() else None,
+            'previous': self.page.previous_page_number() if self.page.has_previous() else None,
+            'count': self.page.paginator.count,
+            'num_pages': self.page.paginator.num_pages,
+            'current_page': self.page.number,
+            'results': data
+        })
```

Comment:
nit: can't you just do this?

```suggestion
        return super().get_paginated_response(data, *args, **kwargs)
```

Reply:
@pomegranited I wanted to replace `next` and `previous` links by page number, but I understand that it was not clear. Updated here: https://github.com/openedx/edx-platform/pull/36311/commits/77cac91211db46dc41b8487fd0d54b02b578a48b

Reply:
Huh ok.. that makes the returned data non-standard for a paginated DRF view, but your update at least makes this clearer.

Reply:
I wanted to make it work in [useInfiniteQuery](https://tanstack.com/query/v4/docs/framework/react/guides/infinite-queries) which expects next or previous page number instead of links.

Reply:
To keep the standard, maybe instead of replacing `next` and `previous`, you can add the page numbers as new values: `next_num_page` and `previous_num_page`. What do you think?

Reply:
@ChrisChV Makes sense, updated.